### PR TITLE
use `towncrier` to handle changelog entries

### PR DESCRIPTION
**Pull request by zacharyburnett**
using `towncrier` to handle changelog entries will
- drastically reduce (or even eliminate) branch conflicts in `CHANGES.rst` we currently experience with PRs
- keep the sections in `CHANGES.rst` consistent and eliminate duplicate sections
- allow automatic links to be generated in the changelog entries

`towncrier` expects "news fragment" files (text files in the `changes/` directory with filenames in the format `<PR#>.<changetype>.rst`, i.e. for this PR it would be `changes/284.general.rst`). See docs at https://towncrier.readthedocs.io/en/latest/tutorial.html#creating-news-fragments

when ready to make a release, run `towncrier build` to ingest the news fragments and generate a changelog section in `CHANGES.rst` with all the new change log entries for that release (this clears the `changes/` directory of all news fragment files). This step should either be done before making a release, or could probably be added to a GitHub workflow triggered on release (to insert a commit and remake the tag).

After merging this PR the `Release Process` wiki page will need to be updated to include a step to run `towncrier build` instead of manually editing `CHANGES.rst`

**Checklist**

- [x] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [ ] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [x] added relevant label(s)

### CODE REVIEW

**Comment by zacharyburnett** [comment]
current running change log in docs:
[image](https://stcal--284.org.readthedocs.build/en/284/stcal/changes.html)

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -14,7 +14,11 @@ This PR addresses ...
...
 
-- [ ] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
+- [ ] for a public change, added a [towncrier news fragment](https://towncrier.readthedocs.io/en/stable/tutorial.html#creating-news-fragments) <details><summary>`changes/<PR#>.<changetype>.rst`</summary>
+
+    - ``changes/<PR#>.apichange.rst``: change to public API
+    - ``changes/<PR#>.bugfix.rst``: fixes an issue
+    - ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
 - [ ] updated relevant tests
 - [ ] updated relevant documentation
 - [ ] updated relevant milestone(s)
```
Comment (zacharyburnett) [inline .github/pull_request_template.md:24]: **Checklist**

 - [ ] for a public change, added a [towncrier news fragment](https://towncrier.readthedocs.io/en/stable/tutorial.html#creating-news-fragments) <details><summary>`changes/<PR#>.<changetype>.rst`</summary>

     - ``changes/<PR#>.apichange.rst``: change to public API
     - ``changes/<PR#>.bugfix.rst``: fixes an issue
     - ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
 - [ ] updated relevant tests
 - [ ] updated relevant documentation
 - [ ] updated relevant milestone(s)

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -14,7 +14,11 @@ This PR addresses ...
...
 
-- [ ] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
+- [ ] for a public change, added a towncrier news fragment in `changes/`: <details><summary>`echo "changed something" > changes/<PR#>.<changetype>.rst`</summary>
+
+    - ``changes/<PR#>.apichange.rst``: change to public API
+    - ``changes/<PR#>.bugfix.rst``: fixes an issue
+    - ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
 - [ ] updated relevant tests
 - [ ] updated relevant documentation
 - [ ] updated relevant milestone(s)
```
Comment (zacharyburnett) [inline .github/pull_request_template.md:24]: **Checklist**

 - [ ] for a public change, added a towncrier news fragment in `changes/`: <details><summary>`echo "changed something" > changes/<PR#>.<changetype>.rst`</summary>

     - ``changes/<PR#>.apichange.rst``: change to public API
     - ``changes/<PR#>.bugfix.rst``: fixes an issue
     - ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
 - [ ] updated relevant tests
 - [ ] updated relevant documentation
 - [ ] updated relevant milestone(s)

File: changes/281.bugfix.rst
Comment (kmacdonald-stsci) [inline changes/281.bugfix.rst:1]: Are these fragments auto-generated or are they something the PR creator needs to create?

**Review by kmacdonald-stsci — APPROVED** [review]

File: changes/281.bugfix.rst
Comment (zacharyburnett) [reply changes/281.bugfix.rst:1]: the PR creator makes them; it's just an `.rst` file containing just the change entry, in the format `<PR#>.<changetype>.rst`, i.e.:
```shell
echo "use towncrier to handle changelog entries" > changes/284.general.rst
```

**Review by braingram — APPROVED** [review]
LGTM!

**Pull request merged**